File: server/webapp/WEB-INF/rails.new/app/presenters/api_v3/config/materials/svn_material_representer.rb
##########################################################################
# Copyright 2017 ThoughtWorks, Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
##########################################################################

module ApiV3
  module Config
    module Materials
      class SvnMaterialRepresenter < ScmMaterialRepresenter
        include ApiV3::Config::Materials::EncryptedPasswordSupport #password property is parsed and encrypted in the module

        property :check_externals
        property :user_name, as: :username
        property :encrypted_password, skip_nil: true, skip_parse: true

      end

    end
  end
end

File: api/api-config-repos-v3/src/main/java/com/thoughtworks/go/apiv3/configrepos/representers/SvnMaterialRepresenter.java

/*
 * Copyright 2019 ThoughtWorks, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.thoughtworks.go.apiv3.configrepos.representers;

import com.thoughtworks.go.api.base.OutputWriter;
import com.thoughtworks.go.api.representers.JsonReader;
import com.thoughtworks.go.config.materials.svn.SvnMaterialConfig;

class SvnMaterialRepresenter implements MaterialRepresenter<SvnMaterialConfig> {

    @Override
    public void toJSON(OutputWriter json, SvnMaterialConfig material) {
        json.add("name", material.getName());
        json.add("auto_update", material.getAutoUpdate());
        json.add("url", material.getUriForDisplay());
        json.add("check_externals", material.isCheckExternals());
        json.add("username", material.getUserName());
        json.addIfNotNull("encrypted_password", material.getEncryptedPassword());
    }

    @Override
    public SvnMaterialConfig fromJSON(JsonReader json) {
        SvnMaterialConfig materialConfig = new SvnMaterialConfig();
        json.readStringIfPresent("name", materialConfig::setName);
        json.readBooleanIfPresent("auto_update", materialConfig::setAutoUpdate);
        json.readStringIfPresent("url", materialConfig::setUrl);
        json.readBooleanIfPresent("check_externals", materialConfig::setCheckExternals);

        json.readStringIfPresent("username", materialConfig::setUserName);

        String password = json.getStringOrDefault("password", null);
        String encryptedPassword = json.getStringOrDefault("encrypted_password", null);
        materialConfig.setEncryptedPassword(PASSWORD_DESERIALIZER.deserialize(password, encryptedPassword, materialConfig));

        return materialConfig;
    }
}
